internal/app/machined/internal/phase/services/post_boot.go

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package services
+
+import (
+	"net"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/talos-systems/talos/internal/app/machined/internal/phase"
+	"github.com/talos-systems/talos/internal/pkg/runtime"
+	"github.com/talos-systems/talos/pkg/config/machine"
+	"github.com/talos-systems/talos/pkg/kubernetes"
+)
+
+// LabelNodeAsMaster represents the LabelNodeAsMaster task.
+type LabelNodeAsMaster struct{}
```

Comment:
nit:  LabelNodeAsMaster -> LabelNodeIfMaster? (since it is safe to be called on any node and will only label it a master _if_ it is a master.